csharp/ql/src/semmle/code/csharp/PrintAst.qll

```diff
@@ -263,8 +310,15 @@ class ControlFlowElementNode extends ElementNode {
     not isNotNeeded(element.getParent+())
   }
 
-  override ElementNode getChild(int childIndex) {
-    result.getElement() = controlFlowElement.getChild(childIndex)
+  override PrintAstNode getChild(int childIndex) {
+    (
+      childIndex = min(int i | exists(controlFlowElement.getChild(i))) - 1
+      or
+      not exists(controlFlowElement.getAChild()) and childIndex = 0
+    ) and
+    result.(TopLevelTypeMentionNode).getTypeMention().getTarget() = controlFlowElement
```

Comment:
It would probably make sense to add a `getTarget()` predicate to `TypeMentionNode` and call that instead (and more places further down).